src/core/PrettyPrinter.js

```diff
@@ -30,6 +30,8 @@ getJasmineRequireObj().pp = function(j$) {
         this.emitScalar('HTMLNode');
       } else if (value instanceof Date) {
         this.emitScalar('Date(' + value + ')');
+      } else if (value && value.toString && typeof value === 'object' && !(value instanceof Array) && value.toString !== Object.prototype.toString) {
```

Comment:
How many of the conditions in this `if` are required to make this functionality work properly? If we have gotten this far down the `else if` chain, we should know that the `value` is not `undefined` or `null`, so at least the first truthy check on `value` should not be necessary.